moda-contracts/src/ModaRegistry.sol

```diff
@@ -39,7 +49,10 @@ contract ModaRegistry is IModaRegistry, AccessControlEnumerable {
     /**
      * @inheritdoc IModaRegistry
      */
-    function setCatalogMembership(uint256 index, address membership) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    function setCatalogMembership(
+        uint256 index,
+        address membership
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
         if (membership == address(0)) {
```

Comment:
This would be a good place to use `supportsInterface()` from ERC165. We would add ERC165 to `IMembership`.

Then we can call `membership.supportsInterface(type(IMembership).interfaceId)`

Reply:
This way we ensure that a user has thought about implementing that interface. Way better than simply checking for a zero address

Reply:
Cool, i have set up like this 

  ```
   function setCatalogMembership(
        uint256 index,
        address membership
    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
        if (!IMembership(membership).supportsInterface(type(IMembership).interfaceId)) {
            revert ContractMustSupportIMembership();
        }

        _catalogs[index].membership = membership;
        emit CatalogMembershipChanged(_catalogs[index].name, membership);
    }
```
This should revert if membership has a supportsInterface function but doesn't implement isMember and also if membership does not have a supportsInterface function right??

Reply:
If a contract implements `supportsInterface` then it will return true or false. It would not revert

Reply:
You should set the `membership` arg type to `IMembership` instead of an `address`. Make it explicit that a user must pass that in. You can probably keep the supportsInterface call you have

Reply:
ahh cool yeah that makes sense, done